osu.Desktop/Program.cs

```diff
@@ -52,15 +52,25 @@ public static void Main(string[] args)
                 // See https://www.mongodb.com/docs/realm/sdk/dotnet/compatibility/
                 if (windowsVersion.Major < 6 || (windowsVersion.Major == 6 && windowsVersion.Minor <= 2))
                 {
-                    // If users running in compatibility mode becomes more of a common thing, we may want to provide better guidance or even consider
-                    // disabling it ourselves.
-                    // We could also better detect compatibility mode if required:
-                    // https://stackoverflow.com/questions/10744651/how-i-can-detect-if-my-application-is-running-under-compatibility-mode#comment58183249_10744730
-                    SDL.SDL_ShowSimpleMessageBox(SDL.SDL_MessageBoxFlags.SDL_MESSAGEBOX_ERROR,
-                        "Your operating system is too old to run osu!",
-                        "This version of osu! requires at least Windows 8.1 to run.\n"
-                        + "Please upgrade your operating system or consider using an older version of osu!.\n\n"
-                        + "If you are running a newer version of windows, please check you don't have \"Compatibility mode\" turned on for osu!", IntPtr.Zero);
+                    bool isInCompatibilityMode = CompatibilityModeChecker.CheckCompatibilityMode();
+
+                    if (isInCompatibilityMode)
+                    {
+                        SDL.SDL_ShowSimpleMessageBox(SDL.SDL_MessageBoxFlags.SDL_MESSAGEBOX_ERROR,
+                            "osu! is running in compatibility mode",
+                            "osu! is running in compatibility mode. This may cause issues with the game. Please ensure osu! is not set to run in compatibility mode.", IntPtr.Zero);
+
+                        CompatibilityModeChecker.LogCompatibilityFlags();
```

Comment:
This won't actually log anything to a logfile as the logger is not yet initialised.

Reply:
Not sure what could be done in this situation.. Logging to console wouldn't be very useful in this case. I could refactor the function to return the flags instead of returning void, then include it in the messagebox. What do you think about that approach?